src/librustc_codegen_ssa/mir/block.rs

```diff
@@ -261,7 +261,10 @@ impl<'a, 'tcx, Bx: BuilderMethods<'a, 'tcx>> FunctionCx<'a, 'tcx, Bx> {
         if self.fn_abi.ret.layout.abi.is_uninhabited() {
             // Functions with uninhabited return values are marked `noreturn`,
             // so we should make sure that we never actually do.
+            // We play it safe by using a well-defined `abort`, and not immediate UB.
             bx.abort();
```

Comment:
I am documenting existing behavior here, but my own inclination would be to remove the `abort` here to give the optimizer some more information. Or should we wait until someone determines that this is actually costing us some optimizations?